### PR TITLE
Feature/5790 participate in cache key generation

### PR DESCRIPTION
**Pull request by HermesSbicego-Laser**
- typo fix #7333

### CODE REVIEW

**Comment by rtpHarry** [comment]
Thanks for your contributions!

It looks like you have opened a second PR for the same thing. When you make a commit to the same branch that you used in the first pull request it automatically updates that pull request.

If you look back at your first PR then you will see the typo commits are in there already so you can close this second PR unless I'm missing some other changes (I've just had a quick glance).

File: src/Orchard.Web/Modules/Orchard.OutputCache/ICachingEventHandler.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿using Orchard.Events;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Web;
+
+namespace Orchard.OutputCache {
+    public interface ICachingEventHandler : IEventHandler {
+        StringBuilder ParticipateInCacheKey(StringBuilder key);
```
Comment (sebastienros) [inline src/Orchard.Web/Modules/Orchard.OutputCache/ICachingEventHandler.cs]: It should be something like `KeyGenerated`. 
It should not return anything.

File: src/Orchard.Web/Modules/Orchard.OutputCache/Filters/OutputCacheFilter.cs [hunk 2]
```diff
@@ -610,6 +613,14 @@ public class OutputCacheFilter : FilterProvider, IActionFilter, IResultFilter, I
                 }
             }
 
+            //make CacheKey morphable by external modules
+            try {
```
Comment (sebastienros) [inline src/Orchard.Web/Modules/Orchard.OutputCache/Filters/OutputCacheFilter.cs]: This is unnecessary, just let it throw if there is an issue, remove the try/catch.

**Comment by sebastienros** [comment]
This is a very limited feature, but we can always add more events later.

**Comment by HermesSbicego-Laser** [comment]
I will apply changes tomorrow.

**Comment by HermesSbicego-Laser** [comment]
Can partially umplement what described in Add events to the Output Cache filter in order to allow modules to extend it #5790

**Pull request merged**